libvast/src/system/archive.cpp

```diff
@@ -185,9 +185,17 @@ archive(archive_type::stateful_pointer<archive_state> self, path dir,
           t.stop(events);
         },
         [=](unit_t&, const error& err) {
-          if (err) {
-            VAST_ERROR(self, "got a stream error:", self->system().render(err));
+          // We get an 'unreachable' error when the stream becomes unreachable
+          // because the actor was destroyed; in this case we can't use `self`
+          // anymore.
+          if (err && err != caf::exit_reason::unreachable) {
+            if (err != caf::exit_reason::user_shutdown)
```

Comment:
I think `user_shutdown` might also be used as a default argument when shutting down the stream, not just when its literally the user shutting down the stream, but I don't completely remember tbh. Did you search for this in the CAF source code?

Reply:
Yes, user_shutdown is just the default exit reason. It doesn't make sense to treat it as an error.